src/qinst/network/NA_N9916A.py

```diff
@@ -0,0 +1,337 @@
+"""
+import numpy as np
+
+from qinst.network_inst import NetworkInst
+
+
+class N9916A(NetworkInst):
+    """N9916A FieldFox Handheld Microwave Analyzer by Keysight."""
+
+    def __init__(
+        self,
+        name: str,
+        address: str,
+        port: int = 5025,
+        timeout: int = 8000,
+        sleep: float = 0.1,
+        no_delay: bool = True,
+        max_points: int = 10001,
+    ):
+        super().__init__(name, address, port, timeout, sleep, no_delay)
+        self._max_points = max_points
+
+    def write_and_hold(self, cmd: str):
+        """Write command and wait until it has been processed."""
+        self.write(cmd)
+        self.hold()
+
+    def clear(self):
+        """Clear the error queue and all status registers."""
+        self.write_and_hold("*CLS")
+
+    def reset(self):
+        """Resetthe device and cancel any pending *OPC command or query."""
+        self.write_and_hold("*RST")
+
+    def hold(self):
+        """Wait until all commands have been processed."""
+        complete = self.query("*OPC?")
+        if complete != "1":
+            raise ValueError(
+                f"Operation completed query returned value {complete} instead of 1."
+            )
+
+    @property
+    def _mode(self):
+        return self.query("INST:SEL?")
+
+    @_mode.setter
+    def _mode(self, mode: str):
+        allowed = ("SA", "NA", "CAT")
+        if mode in allowed:
+            return self.write_and_hold(f'INST:SEL "{mode}"')
+        else:
+            raise ValueError(f"Invalid mode selected, choose between {allowed}.")
+
+    @property
+    def f_min(self):
+        """Minimum frequency."""
+        return float(self.query("SENS:FREQ:START?"))
+
+    @f_min.setter
+    def f_min(self, f: float):
+        self.write(f"FREQ:START {abs(f)}")
+
+    @property
+    def f_max(self):
+        """Maximum frequency."""
+        return float(self.query("SENS:FREQ:STOP?"))
+
+    @f_max.setter
+    def f_max(self, f: float):
+        self.write(f"FREQ:STOP {abs(f)}")
+
+    @property
+    def f_center(self):
+        """Central frequency."""
+        return float(self.query("SENS:FREQ:CENT?"))
+
+    @f_center.setter
+    def f_center(self, f: float):
+        self.write(f"SENS:FREQ:CENT {abs(f):5.6f}")
+
+    @property
+    def f_span(self):
+        """Frequency span."""
+        return float(self.query("FREQ:SPAN?"))
+
+    @property
+    def sweep_points(self):
+        """Number of points in sweep."""
+        return int(self.query("SENS:SWE:POIN?"))
+
+    @sweep_points.setter
+    def sweep_points(self, npoints):
+        npoints = min(abs(npoints), self._max_points)
+        self.write(f"SWE:POIN {npoints}")
+
+    @property
+    def continuous(self):
+        """Acquisition mode."""
+        return self.query("INIT:CONT?")
+
+    @continuous.setter
+    def continuous(self, status: bool):
+        self.write_and_hold(f"INIT:CONT {int(status)}")
+
+    def single_sweep(self):
```

Comment:
Non mi è chiaro cosa faccia questa funzione... Fa uno sweep in frequenza/ampiezza ma non salva/restituisce alcun dato giusto? Cosa serve?